### PR TITLE
Update fantomas to v6

### PR DESCRIPTION
**Pull request by dawedawe**
This updates fantomas to v6 and adjusts the .editorconfig

### CODE REVIEW

File: ApiSurface.DocumentationSample/KitchenSink.fs [hunk 1]
```diff
@@ -200,23 +200,23 @@ type internal MomentActionsId =
 type IndexedProperties<'a, 'b> =
     /// Simple getter with one argument
     member this.Item
-        with get (_ : int) : string = failwith ""
```
Comment (nojaf) [inline ApiSurface.DocumentationSample/KitchenSink.fs:203]: This one is interesting actually.
As `fsharp_space_before_colon=true` is listed in the settings we no longer seem to respect this.
This is most likely a regression after Dallas.
Could you try and find out what happened here?

Comment (dawedawe) [reply ApiSurface.DocumentationSample/KitchenSink.fs:203]: Sure, I'll have a look.

File: .git-blame-ignore-revs [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Fantomas 6.0.0-alpha-010
+c9d0d84ff5b82641deb1a7d315dfa92307b7997b
```
Comment (Smaug123) [inline .git-blame-ignore-revs:2]: By the way, since this repository requires squash-merging, the only way to do this is to create another PR once this one goes in, listing the hash of the squash commit.

Comment (dawedawe) [reply .git-blame-ignore-revs:2]: Ah yeah, thanks for the heads-up. Will do.

Comment (Smaug123) [reply .git-blame-ignore-revs:2]: Could you please revert the change to this file, given that it won't do anything? (Assuming you're actually proposing this PR and it isn't just to see how Fantomas v6 behaves.)

Comment (dawedawe) [reply .git-blame-ignore-revs]: done

**Review by Smaug123 — DISMISSED** [review]

**Review by nojaf — DISMISSED** [review]

**Comment by dawedawe** [comment]
Although this seems to be fine and it will most likely be just a change of names, we would like to wait for the final v6 release for this to get merged. Shouldn't be too far in the future.

**Review by Smaug123 — APPROVED** [review]

**Review by nojaf — APPROVED** [review]
